### PR TITLE
EVM Wallets on Substrate

### PR DESCRIPTION
**Pull request by onnovisser**
### Description

Adds basic EVM support, behind a debug flag. Left todo is the option to have separate signing and sending of transactions, which is needed for creating loans. 
Also upgrades to WalletConnect V2

### Approvals

- [ ] Dev

### CODE REVIEW

File: centrifuge-js/src/utils/index.ts [hunk 1]
```diff
@@ -141,3 +141,8 @@ export function computeMultisig(multisig: Multisig): ComputedMultisig {
     threshold: multisig.threshold,
   }
 }
+
+export function evmToSubstrateAddress(address: string) {
```
Comment (branan) [inline centrifuge-js/src/utils/index.ts:145]: We should probably make this an RPC/runtime API so that we can update the algorithm more easily

**Comment by branan** [comment]
I'm really impressed how easily this all came together. `wrapSubstrateEvmSignAndSend` looks pretty much how I would expect it to. The rest of the frontend bits (especially the wallet component) are a bit out of my area of expertise to comment on

File: centrifuge-react/src/components/WalletProvider/WalletProvider.tsx [hunk 2]
```diff
@@ -100,16 +135,44 @@ export function WalletProvider({
...
+  const consts = useCentrifugeConsts()
+  const centEvmChainId = useCentEvmChainId()
+
+  const evmChains = React.useMemo(() => {
+    const chains = {
+      ...evmChainsProp,
+    }
+    if (centEvmChainId) {
+      chains[centEvmChainId] = {
+        urls: ['https://fullnode.development.cntrfg.com/'],
```
Comment (sophialittlejohn) [inline centrifuge-react/src/components/WalletProvider/WalletProvider.tsx:152]: this should be passed down as an env variable

**Comment by sophialittlejohn** [comment]
nice job @onnovisser!

I managed to get some DEVEL and aUSD into my evm wallet. From the screenshot it appears I have enough of both DEVEL and aUSD to invest.

Screenshot 2023-06-29 at 16 36 33

The tx fails though stating that my balance is too low 

Screenshot 2023-06-29 at 16 51 57

**Review by sophialittlejohn — APPROVED** [review]
🥳

**Pull request merged**